src/realm/array_decimal128.hpp

```diff
@@ -104,6 +104,72 @@ class ArrayDecimal128 : public ArrayPayload, private Array {
     }
 };
 
+template <>
+class QueryStateMin<Decimal128> : public QueryStateBase {
+public:
+    QueryStateMin(size_t limit = -1)
+        : QueryStateBase(limit)
+    {
+        m_state = Decimal128("+inf");
+    }
+    bool match(size_t index, Mixed value) override
+    {
+        if (!value.is_null()) {
+            ++m_match_count;
+            if (value.get<Decimal128>() < m_state) {
+                m_state = value.get<Decimal128>();
+                if (m_key_values) {
+                    m_minmax_index = m_key_values->get(index) + m_key_offset;
+                }
+                else {
+                    m_minmax_index = int64_t(index);
+                }
+            }
+        }
+        return (m_limit > m_match_count);
+    }
+    Decimal128 get_min() const
+    {
+        return m_match_count ? m_state : Decimal128{};
+    }
+
+private:
+    Decimal128 m_state;
+};
+
+template <>
+class QueryStateMax<Decimal128> : public QueryStateBase {
+public:
+    QueryStateMax(size_t limit = -1)
+        : QueryStateBase(limit)
+    {
+        m_state = Decimal128("-inf");
+    }
+    bool match(size_t index, Mixed value) override
```

Comment:
It seems error-prone that these implementations are more or less copy-pasted for each type. Could we not use a template for that?

Reply:
Indeed. I did some cleanup.